src/software/thunderscope/replay/proto_player.py

```diff
@@ -85,22 +109,138 @@ def __sort_replay_chunks(file_path: str):
             replay_index, _ = tail.split(".")
             return int(replay_index)
 
-        self.sorted_chunks = sorted(replay_files, key=__sort_replay_chunks)
+        return sorted(replay_files, key=__sort_replay_chunks)
+
+    @staticmethod
+    def is_valid_protobuf(message):
+        """
+        True if the time send is greater than 2023 unix timestamp, and the protobuf have the timestamp field
+
+        :return: True if the time send is greater than 2023 unix timestamp, and the protobuf have the timestamp field
+        """
+        unixtimestamp_2023 = datetime(year=2023, month=1, day=1).timestamp()
 
-        # We can get the total runtime of the log from the last entry in the last chunk
-        last_chunk_data = ProtoPlayer.load_replay_chunk(self.sorted_chunks[-1])
         try:
-            self.end_time, _, _ = ProtoPlayer.unpack_log_entry(last_chunk_data[-1])
-        except DecodeError:
-            self.end_time, _, _ = ProtoPlayer.unpack_log_entry(last_chunk_data[-2])
-        logging.info(
-            "Loaded log file with total runtime of {:.2f} seconds".format(self.end_time)
+            if message.time_sent.epoch_timestamp_seconds > unixtimestamp_2023:
+                # has the time send field and is greater than 2023 unix timestamp
+                return True
+
+            # return false where the timestamp is way to small!
+            return False
+
+        # does not contain the time_sent field, so we have a type error
+        except AttributeError:
+            return False
+
+    def convert_field_test_replayfiles(self):
```

Comment:
it's a bit weird to rewrite the log files.

I feel it makes more sense to write to a new replay folder instead of corrupting a folder that you may be reading

Reply:
Personally, Im leaning more towards not manually checking if a replay file is from a field test by checking for hints of old bugs + recreating the file. I think we can ignore the previous bugs in ProtoLogger and ignore having custom conversions for previously corrupted replay files, and instead focus on ways of fixing these bugs (like what we have discussed above). In short, I dont think we should have these new methods in this class.